lms/djangoapps/badges/models.py

```diff
@@ -0,0 +1,328 @@
+            # Fall back to default, if there is one.
+            return cls.objects.get(default=True).icon
+
+    class Meta(object):
+        app_label = "badges"
+
+
+class CourseEventBadgesConfiguration(ConfigurationModel):
+    """
+    Determines the settings for meta course awards-- such as completing a certain
+    number of courses or enrolling in a certain number of them.
+    """
+    courses_completed = models.TextField(
+        blank=True, default='',
+        help_text=_(
+            u"On each line, put the number of completed courses to award a badge for, a comma, and the slug of a "
+            u"badge class you have created that has the issuing component 'openedx__course'. "
+            u"For example: 3,enrolled_3_courses"
+        )
+    )
+    courses_enrolled = models.TextField(
+        blank=True, default='',
+        help_text=_(
+            u"On each line, put the number of enrolled courses to award a badge for, a comma, and the slug of a "
+            u"badge class you have created that has the issuing component 'openedx__course'. "
+            u"For example: 3,enrolled_3_courses"
+        )
+    )
+    course_groups = models.TextField(
+        blank=True, default='',
+        help_text=_(
+            u"Each line is a comma-separated list. The first item in each line is the slug of a badge class you "
+            u"have created that has an issuing component of 'openedx__course'. The remaining items in each line are "
+            u"the course keys the learner needs to complete to be awarded the badge. For example: "
+            u"slug_for_compsci_courses_group_badge,course-v1:CompSci+Course+First,course-v1:CompsSci+Course+Second"
+        )
+    )
+
+    def __unicode__(self):
+        return u"<CourseEventBadgesConfiguration ({})>".format(u"Enabled" if self.enabled else u"Disabled")
+
+    @staticmethod
+    def get_specs(text):
+        """
+        Takes a string in the format of:
+            int,course_key
+            int,course_key
+
+        And returns a dictionary with the keys as the numbers and the values as the course keys.
+        """
+        specs = text.splitlines()
+        specs = [line.split(',') for line in specs if line.strip()]
+        return {int(num): slug.strip().lower() for num, slug in specs}
+
+    @property
+    def completed_settings(self):
+        """
+        Parses the settings from the courses_completed field.
+        """
+        return self.get_specs(self.courses_completed)
+
+    @property
+    def enrolled_settings(self):
+        """
+        Parses the settings from the courses_completed field.
+        """
+        return self.get_specs(self.courses_enrolled)
+
+    @property
+    def course_group_settings(self):
+        """
+        Parses the course group settings. In example, the format is:
+
+        slug_for_compsci_courses_group_badge,course-v1:CompSci+Course+First,course-v1:CompsSci+Course+Second
+        """
+        specs = self.course_groups.strip()
```

Comment:
This implementation seems similar to the work being done in get_specs above -- is there an opportunity for reuse here?